src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -123,10 +126,10 @@ client_ptr kafka_client_cache::fetch_or_insert_impl(
         inner_list.relocate(inner_list.begin(), it_list);
     }
 
-    return client;
+    return std::make_pair(client, client_sm);
 }
 
-client_ptr kafka_client_cache::fetch_or_insert(
+std::pair<client_ptr, client_sm_ptr> kafka_client_cache::fetch_or_insert(
```

Comment:
This is a bit of a leaky abstraction now. I wonder if it should be private.

Reply:
This is a nit and can be handled later.

Reply:
Done but I made it protected so I can still test `fetch_or_insert` in the unit tests.